staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go

```diff
@@ -72,8 +72,22 @@ func IsNotMarshalable(err error) bool {
 // is passed, the encoded object will have group, version, and kind fields set. If typer is nil, the objects will be written
 // as-is (any type info passed with the object will be used).
 func NewSerializer(creater runtime.ObjectCreater, typer runtime.ObjectTyper) *Serializer {
+	return newSerializer(creater, typer, protoEncodingPrefix)
+}
+
+// NewSerializerWithPrefix creates a Protobuf serializer that handles encoding versioned objects into the proper wire form. If a typer
+// is passed, the encoded object will have group, version, and kind fields set. If typer is nil, the objects will be written
+// as-is (any type info passed with the object will be used). If no encoding prefix is passed, the default encoding prefix will be used.
+func NewSerializerWithPrefix(creater runtime.ObjectCreater, typer runtime.ObjectTyper, encodingPrefix []byte) *Serializer {
+	if encodingPrefix == nil {
+		encodingPrefix = protoEncodingPrefix
+	}
+	return newSerializer(creater, typer, encodingPrefix)
+}
+
+func newSerializer(creater runtime.ObjectCreater, typer runtime.ObjectTyper, encodingPrefix []byte) *Serializer {
```

Comment:
This seems kind of odd. I'd recommend not having a default. If no prefix is invalid (which I'm not sure it should be), you should throw an error.